app/Http/Controllers/AuthController.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Hash;
+use App\Models\User;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer\Hmac\Sha256;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use App\Resources\UserResource;
+
+class AuthController extends Controller {
+    private function jwt(User $user) {
+        $config = Configuration::forSymmetricSigner(new Sha256(), InMemory::plainText(env('APP_KEY')));
+        $now = new \DateTimeImmutable();
+        $token = $config->builder()
+            ->issuedBy(env('APP_URL'))
+            ->permittedFor(env('APP_URL'))
+            ->identifiedBy(uniqid())
+            ->issuedAt($now)
+            ->canOnlyBeUsedAfter($now)
+            ->expiresAt($now->modify('+'.env('JWT_EXPIRE')))
```

Comment:
```suggestion
            ->expiresAt($now->copy()->modify('+'.env('JWT_EXPIRE')))
```
nowが今後使われない暗黙の了解に依存してるのも嫌なので

Reply:
immutableマジ?時代は進歩してるのね…老害と申します